lib/Importer/TFLiteModelLoader.cpp

```diff
@@ -1968,152 +1983,320 @@ Error TFLiteModelLoader::loadSlice(const tflite::Operator *op,
-  llvm::SmallVector<NodeValue, 4> inputs;
-  inputs.reserve(numInputs);
-  for (size_t idx = 0; idx < numInputs; ++idx) {
-    NodeValue input;
-    ASSIGN_VALUE_OR_RETURN_ERR(input, getInputNodeValue(op, idx));
-    inputs.push_back(input);
+    auto inDims = input.getType()->dims();
+
+    // Create Padding Node.
+    auto padsH = padsC->getPayload().getHandle<int32_t>();
+    std::vector<int> padsVec(2 * inDims.size());
+    for (int32_t i = 0; i < blockSize; i++) {
+      padsVec[i + 1] = padsH.raw(2 * i + 0);
+      padsVec[i + 1 + inDims.size()] = padsH.raw(2 * i + 1);
+    }
+    ShapeVector padDims(inDims.begin(), inDims.end());
+    for (dim_t i = 0; i < padDims.size(); i++) {
+      padDims[i] += (padsVec[i] + padsVec[i + inDims.size()]);
+    }
+    auto OT = mod_.uniqueTypeWithNewShape(input.getType(), padDims);
+    NodeValue pad = F_->createPad(opInfo.name, input, OT, PaddingMode::CONSTANT,
+                                  padsVec, 0.f);
+
+    // Check expected dimensions.
+    auto padInDims = pad.getType()->dims();
+    auto outDims = outTy->dims();
+    int32_t calcblockSize = 0;
+    for (int32_t i = 0; i < blockSize; i++) {
+      RETURN_ERR_IF_NOT(
+          padInDims[i + 1] % outDims[i + 1] == 0,
+          opErrMsg(opInfo,
+                   strFormat("Input value %d must be a multiple of "
+                             "output value %d for space dim %d",
+                             (int32_t)padInDims[1], (int32_t)outDims[1], i)));
+      int32_t newBlockSize = padInDims[i + 1] / outDims[i + 1];
+      if (i > 1) {
+        RETURN_ERR_IF_NOT(
+            calcblockSize == newBlockSize,
+            opErrMsg(opInfo, "Block Size for H & W must be the same."));
+        calcblockSize = newBlockSize;
+      }
+    }
+
+    // Transpose N and D and perform SpaceToDepth.
+    auto *trIn = F_->createTranspose(opInfo.name + ".trIn", pad, {3, 1, 2, 0});
+    auto *S2D = F_->createSpaceToDepth(opInfo.name + ".S2D", trIn, blockV[0]);
+    auto *output =
+        F_->createTranspose(opInfo.name + ".trOut", S2D, {3, 1, 2, 0});
+
+    return setOutputNodeValue(op, output);
   }
 
-  unsigned_t axis;
-  ASSIGN_VALUE_OR_RETURN_ERR(
-      axis, getPositiveAxis<unsigned_t>(opts->axis(), outTy->dims().size()));
+  Error TFLiteModelLoader::loadBatchToSpaceNd(const tflite::Operator *op,
+                                              const OperatorInfo &opInfo) {
+    NodeValue input;
+    ASSIGN_VALUE_OR_RETURN_ERR(input, getInputNodeValue(op, 0));
+    NodeValue block;
+    ASSIGN_VALUE_OR_RETURN_ERR(block, getInputNodeValue(op, 1));
+    NodeValue crop;
+    ASSIGN_VALUE_OR_RETURN_ERR(crop, getInputNodeValue(op, 2));
+    TypeRef outTy;
+    ASSIGN_VALUE_OR_RETURN_ERR(outTy, getOutputType(op, 0));
 
-  // Validate that all inputs have same shape.
-  for (size_t idx = 0; idx < numInputs; ++idx) {
+    // Implemented as Transpose N-D -> DepthToSpace -> Transpose N-D -> Crop
```

Comment:
Could you create a Graph.cpp utility for creating this sequence of nodes e.g. ```Node *createBatchToSpaceNd(...))```.
This makes the logic reusable but also this will allow us later (if necessary) to add a ```BatchToSpaceNdNode``` and a lowering logic for it.